### ISSUE TITLE
Destroying MCA Subscription created with `azurerm_subscription` take a long time.

### ISSUE THREAD

**Issue by donwlewis** [terraform-providers/terraform-provider-azurerm#11192]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.14.7
+ provider registry.terraform.io/hashicorp/azurerm v2.53.0

### Affected Resource(s)

* `azurerm_subscription`

### Terraform Configuration Files

```hcl
data "azurerm_billing_mca_account_scope" "mca" {
  billing_account_name = REDACTED
  billing_profile_name = REDACTED
  invoice_section_name = REDACTED
}

resource "azurerm_subscription" "sub" {
  subscription_name = "test-subscription"
  billing_scope_id  = data.azurerm_billing_mca_account_scope.mca.id
}
```

### Expected Behaviour

Terraform should finish the destroy in a fairly timely window or timeout.

### Actual Behaviour

Terraform is destroying 2+ hours later.

### Steps to Reproduce

1. `terraform apply`
2. Verify subscription was successfully created.
3. `terraform destroy`

### Important Factoids

I am writing this issue as the destroy process is going, and it has been 2+ hours.  I am not sure its going to time out or finish.  I will update as I discover more.

**Comment by favoretti** [comment]
Hi there and thank you for reporting this. From the looks of it #11124 should have fixed that. Can you please retry with provider `2.54.0`? Thank you.

**Comment by donwlewis** [comment]
This seems fixed in version `2.54.0`.  I tried to look for an already existing issue for this, but didn't find it, so I apologize for the dupe.

**Comment by favoretti** [comment]
No worries. Thank you for confirming. Closing this one now. Feel free to reopen or open a new one if the problem persists.
